features/options/extract/turn_function.feature

```diff
@@ -0,0 +1,155 @@
+@routing @testbot @turn_function
+Feature: Turn Function Information
+
+    Scenario: Turns should have correct information of source and target
+        Given the profile file
+        """
+        functions = require('feature/car_turnbot')
+
+        return {
+          setup = functions.setup,
+          process_way = functions.process_way,
+          process_node = functions.process_node,
+          process_turn = functions.process_turn
+        }
+
+        """
+
+        And the node map
+            """
+
+            a b c
+
+            """
+        And the ways
+            | nodes | highway  |
+            | ab    | motorway |
+            | bc    | motorway |
+        And the data has been saved to disk
+
+        When I run "osrm-extract --profile {profile_file} {osm_file}"
+        Then it should exit successfully
+        And stdout should contain "source_is_motorway true"
+        And stdout should contain "target_is_motorway true"
+        And stdout should contain "source_is_link false"
+        And stdout should contain "source_speed 90"
+        And stdout should contain "target_is_motorway true"
+        And stdout should contain "target_is_link false"
+        And stdout should contain "target_speed 90"
+
+
+    Scenario: Turns should detect when turn is leaving highway
+        Given the profile file
+        """
+        functions = require('feature/car_turnbot')
+
+        return {
+          setup = functions.setup,
+          process_way = functions.process_way,
+          process_node = functions.process_node,
+          process_turn = functions.process_turn
+        }
+
+
+        """
+
+        And the node map
+            """
+
+            a b c
+
+            """
+        And the ways
+            | nodes | highway       | lanes |
+            | ab    | motorway      | 3     |
+            | bc    | motorway_link |       |
+
+        And the data has been saved to disk
+
+        When I run "osrm-extract --profile {profile_file} {osm_file}"
+        Then it should exit successfully
+        And stdout should contain "source_is_motorway true"
+        And stdout should contain "source_is_link false"
+        And stdout should contain "source_speed 90"
+        And stdout should contain "source_number_of_lanes 3"
+        And stdout should contain "target_is_motorway false"
+        And stdout should contain "target_is_link true"
+        # Question @review: should number_of_lanes when untagged be 0 or 1?
```

Comment:
:+1: for using 0 number of lanes when untagged, otherwise `number_of_lanes=1` would mean both untagged lanes and one lane cases.